### PR TITLE
Add forum abilities

### PR DESCRIPTION
**Pull request by xzhflying**

### CODE REVIEW

File: app/models/components/course/forums_ability_component.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    can :manage, Course::Forum, course_staff_hash
+  end
+
+  def allow_staff_manage_topics
+    can :manage, Course::Forum::Topic, topic_course_staff_hash
+
+    can :set_sticky, Course::Forum::Topic, topic_course_staff_hash
+    can :set_announcement, Course::Forum::Topic, topic_course_staff_hash
+  end
+end
```
Comment (xzhflying) [inline app/models/components/course/forums_ability_component.rb]: fixed

**Comment by lowjoel** [comment]
Wait for #630. I told @allenwq to remove the `require 'cancancan/matchers'` frome ach spec and into rails_helper

**Comment by xzhflying** [comment]
OK!

**Comment by lowjoel** [comment]
Remove the matchers

**Comment by xzhflying** [comment]
Done!

File: app/models/components/course/forums_ability_component.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+  def allow_staff_manage_forums
+    can :manage, Course::Forum, course_staff_hash
+  end
+
+  def allow_staff_manage_topics
+    can :manage, Course::Forum::Topic, topic_course_staff_hash
+
+    can :set_sticky, Course::Forum::Topic, topic_course_staff_hash
+    can :set_announcement, Course::Forum::Topic, topic_course_staff_hash
```
Comment (allenwq) [inline app/models/components/course/forums_ability_component.rb]: These are not needed, they are covered by `manage`

File: app/models/components/course/forums_ability_component.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    can :read, Course::Forum::Topic, topic_all_course_users_hash.reverse_merge(hidden: false)
+    can :subscribe, Course::Forum::Topic, topic_all_course_users_hash.reverse_merge(hidden: false)
+  end
+
+  def allow_students_create_topics
+    can :create, Course::Forum::Topic, topic_all_course_users_hash
+  end
+
+  def allow_students_update_topics
+    can :edit, Course::Forum::Topic, topic_all_course_users_hash.reverse_merge(hidden: false,
```
Comment (allenwq) [inline app/models/components/course/forums_ability_component.rb]: This is also not needed, `:update` is the alias of `:edit` and `:update`

**Comment by xzhflying** [comment]
@lowjoel

**Pull request merged**